src/routes/auditlog/+page.svelte

```diff
@@ -0,0 +1,371 @@
+<script lang="ts">
+	import LayoutSearchFilter from '$lib/comp/LayoutSearchFilter.svelte';
+	import { UNKNOWN_ITEM } from '$lib/const.js';
+	import Checkbox from '$lib/controls/Checkbox.svelte';
+	import {
+		evaluateLogicalStringSearch,
+		isOnlyDigits,
+		logicalSearchTooltip,
+		onlyUnique,
+		properUrlEncode,
+		titleCase
+	} from '$lib/util.js';
+	import { AuditLog, Prisma } from '@prisma/client';
+	import { onMount } from 'svelte';
+
+	const dateOptions: Intl.DateTimeFormatOptions = {
+		year: 'numeric',
+		month: 'short',
+		day: 'numeric',
+		hour: '2-digit',
+		minute: '2-digit'
+	};
+
+	export let data;
+	let logs: (AuditLog & { linkable: boolean; mission: string | null })[] = data.logs;
+
+	let logRows: any = {};
+	let resultsText: number = logs.length;
+	let searchOptionBoxes = ['from user', 'mission', 'mission pack', 'solve', 'user', 'invert'];
+	let searchOptionTooltips = [
+		'Search for the user that made the change',
+		'Search for mission name',
+		'Search for mission pack name',
+		'Search for solve team or mission name for a solve',
+		'Search for edited users',
+		'Invert search'
+	];
+	let layoutSearch: LayoutSearchFilter;
+	let searchOptions: string[] = [];
+	let showAll = false;
+	const resultLimit = 50;
+	let validSearchOptions: boolean[] = Array(searchOptionBoxes.length).fill(false);
+	validSearchOptions[1] = validSearchOptions[2] = validSearchOptions[3] = true;
+
+	function pastTense(log: AuditLog): string {
+		if (
+			(log.model === 'Mission' || log.model === 'MissionPack' || log.model === 'Completion') &&
+			(log.action === 'delete' || log.action === 'update')
+		) {
+			const statsBef = JSON.parse(JSON.stringify(log.before));
+			const statsAft = JSON.parse(JSON.stringify(log.after));
+			if (log.action === 'delete' && statsBef.verified === false) return 'rejected';
+			else if (log.action === 'update' && statsBef.verified === false && statsAft.verified === true) return 'accepted';
+		}
+		return log.action + (log.action.endsWith('e') ? 'd' : 'ed');
+	}
+	function shortName(str: string): string {
+		if (str.length > 24) return str.substring(0, 22) + ' ...';
+		else return str;
+	}
+	function unlinkable(str: string): string {
+		if (str === UNKNOWN_ITEM) return 'Unknown';
+		return shortName(str);
+	}
+	function shortTeam(str: string): string {
+		if (str.length > 23) return str.substring(0, 21) + '...';
+		else return str;
+	}
+	function shortItem(str: string): string {
+		if (str.length > 13) return str.substring(0, 11) + '...';
+		else return str;
+	}
```

Comment:
In case you didn't already look into this, there is a text-overflow property in css that does this: https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow